core/src/usage/lifetimes.rs

```diff
@@ -275,15 +294,15 @@ impl UsesLifetimes for syn::TypeParamBound {
         match *self {
             syn::TypeParamBound::Trait(ref v) => v.uses_lifetimes(options, lifetimes),
             syn::TypeParamBound::Lifetime(ref v) => v.uses_lifetimes(options, lifetimes),
+            _ => Default::default(),
```

Comment:
Returning `default` here feels odd, since it's going to cause us to silently miss type params in new syntax constructs.